File: apps/admin/src/api/secret.ts
import { apiFetch } from './http'
import { buildListQuery } from './query'
import type {
  SecretCreateRep,
  SecretCreateReq,
  SecretListRep,
  SecretListReq,
  SecretMain,
  SecretUpdateReq,
} from './types'

/**
 * REST client for the kusec `Secret` service (see api/proto/kusec_v1/secret.proto).
 * Routes are singular per the gateway contract: `/secret`, `/secret/{id}`.
 * A Secret is scoped to an App via `app_id`.
 */

export function listSecrets(req: SecretListReq = {}): Promise<SecretListRep> {
  const query = buildListQuery(req.list_params, {
    app_id: req.app_id,
    active: req.active,
    search: req.search,
  })
  return apiFetch<SecretListRep>(`/secret${query}`)
}

export function getSecret(id: string): Promise<SecretMain> {
  return apiFetch<SecretMain>(`/secret/${encodeURIComponent(id)}`)
}

export function createSecret(req: SecretCreateReq): Promise<SecretCreateRep> {
  return apiFetch<SecretCreateRep>('/secret', {
    method: 'POST',
    body: JSON.stringify(req),
  })
}

export function updateSecret(id: string, req: SecretUpdateReq): Promise<void> {
  return apiFetch<void>(`/secret/${encodeURIComponent(id)}`, {
    method: 'PUT',
    body: JSON.stringify(req),
  })
}

export function deleteSecret(id: string): Promise<void> {
  return apiFetch<void>(`/secret/${encodeURIComponent(id)}`, {
    method: 'DELETE',
  })
}

File: internal/mcpserver/client/model/secret.go
package model

type Secret struct {
	ID             string `json:"id"`
	CreatedAt      string `json:"created_at"`
	UpdatedAt      string `json:"updated_at"`
	AppID          string `json:"app_id"`
	Active         bool   `json:"active"`
	SlugName       string `json:"slug_name"`
	Description    string `json:"description"`
	KubeSecretName string `json:"kube_secret_name"`
	KubeType       string `json:"kube_type"`
	ExactSlug      bool   `json:"exact_slug"`
}

// List

type SecretListReq struct {
	ListParams ListParams
	AppID      *string
	Active     *bool
	Search     *string
}

type SecretListRep struct {
	PaginationInfo PaginationInfo `json:"pagination_info"`
	Results        []Secret       `json:"results"`
}

// Create

type SecretCreateReq struct {
	AppID       string `json:"app_id"`
	Active      *bool  `json:"active,omitempty"`
	SlugName    string `json:"slug_name"`
	Description string `json:"description"`
	KubeType    string `json:"kube_type,omitempty"`
}

type SecretCreateRep struct {
	ID string `json:"id"`
}

// Update

type SecretUpdateReq struct {
	Active      *bool   `json:"active,omitempty"`
	SlugName    *string `json:"slug_name,omitempty"`
	Description *string `json:"description,omitempty"`
	KubeType    *string `json:"kube_type,omitempty"`
}
